examples/src/main/scala/TraceExample.scala

```diff
@@ -133,17 +132,23 @@ object TraceExample extends IOApp.Simple {
             InstitutionServiceClient.apply[IO],
             UserDatabase.apply[IO]
           )
-          val resource: Resource[IO, Unit] =
-            Resource.make(IO.sleep(50.millis))(_ => IO.sleep(100.millis))
           tracer
-            .resourceSpan("Start up")(resource)
-            .surround(
-              userIdAlg
-                .getAllUsersForInstitution(
-                  "9902181e-1d8d-4e00-913d-51532b493f1b"
-                )
-                .flatMap(IO.println)
-            )
+            .spanResource("Start up")
+            .use { case (_, nt /* natural transformation */ ) =>
+              for {
+                _ <- nt(tracer.span("acquire").surround(IO.sleep(50.millis)))
+                _ <- nt {
+                  tracer.span("use").surround {
+                    userIdAlg
+                      .getAllUsersForInstitution(
+                        "9902181e-1d8d-4e00-913d-51532b493f1b"
+                      )
+                      .flatMap(IO.println)
+                  }
+                }
+                _ <- nt(tracer.span("release").surround(IO.sleep(100.millis)))
+              } yield ()
+            }
```

Comment:
I'm concerned that my original commit implemented this incorrectly (before I'd quite understood things), but nothing caught it. is there a way we can run the examples and make sure they're correct?

Reply:
Hmm.  sbt-doctest can make unit tests out of scaladoc comments, but I don't know of anything that tests the output of examples.